src/command/attributecommand.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import Command from './command.js';
+import TreeWalker from '../treemodel/treewalker.js';
+import Range from '../treemodel/range.js';
+import Selection from '../treemodel/selection.js';
+
+/**
+ * An extension of basic {@link core.command.Command} class, which provides utilities for a command that sets a single
+ * attribute on a text or element with value `true`. AttributeCommand uses {@link core.treeModel.Document#selection} to
+ * decide which nodes (if any) should be changed, and applies or removes attributes from them.
+ * See {@link core.treeView.Converter#execute} for more.
+ *
+ * The command checks {@link core.treeModel.Document#schema} to decide if it should be enabled.
+ * See {@link core.treeView.Converter#checkSchema} for more.
+ *
+ * @memberOf core.command
+ */
+export default class AttributeCommand extends Command {
```

Comment:
I wonder which pieces of this command could be useful outside it. Right now we assume that to operate on attributes you need this command, but what if your functionality needs to perform similar action but without registering a command?

I see that either we can expose some of this command's functionality into some `AttrbiuteApplier` or we can tell – you want a piece of this command, then create it (what can be done without registering).